src/main/java/io/cdap/plugin/successfactors/source/metadata/SuccessFactorsSchemaGenerator.java

```diff
@@ -0,0 +1,770 @@
+      edmAnnotationAttributes.forEach(sapAttribute -> {
+        String sapAttributeText = sapAttribute.getText();
+        switch (sapAttribute.getName()) {
+          case SapAttributes.DISPLAY_FORMAT:
+            successFactorsColumnDetailBuilder.displayFormat(sapAttributeText);
+            break;
+          case SapAttributes.FILTER_RESTRICTION:
+            successFactorsColumnDetailBuilder.filterRestrictions(sapAttributeText);
+            break;
+          case SapAttributes.REQUIRED_IN_FILTER:
+            successFactorsColumnDetailBuilder.requiredInFilter(Boolean.parseBoolean(sapAttributeText));
+            break;
+          case SapAttributes.LABEL:
+            successFactorsColumnDetailBuilder.label(sapAttributeText);
+            break;
+        }
+      });
+    }
+
+    if (!edmProperty.isSimple()) {
+      List<SuccessFactorsColumnMetadata> complexChild = buildComplexTypes(namespace, edmProperty);
+      if (!complexChild.isEmpty()) {
+        successFactorsColumnDetailBuilder.childList(complexChild);
+      }
+    }
+
+    return successFactorsColumnDetailBuilder.build();
+  }
+
+
+  /**
+   * Build schema with the all the default (non-navigation) properties for given entity name along with
+   * all the navigation property provided in the 'expandOption'.
+   *
+   * @param entityName   service entity name
+   * @param expandOption all the selective expanded property names
+   * @return {@code Schema}
+   * @throws SuccessFactorsServiceException throws in following two cases
+   *        1. if neither default nor expanded property were found in the given entity name,
+   *        2. if fails at apache olingo processing.
+   */
+  public Schema buildExpandOutputSchema(String entityName, String expandOption) throws SuccessFactorsServiceException {
+    try {
+      List<SuccessFactorsColumnMetadata> columnDetailList = buildDefaultColumns(entityName);
+      if (columnDetailList.isEmpty()) {
+        throw new SuccessFactorsServiceException(ResourceConstants.ERR_NO_COLUMN_FOUND.getMsgForKey(DEFAULT_PROPERTY,
+          entityName));
+      }
+
+      List<SuccessFactorsColumnMetadata> expandColumnDetailList = buildExpandedEntity(entityName, expandOption);
+      if (expandColumnDetailList.isEmpty()) {
+        throw new SuccessFactorsServiceException(ResourceConstants.ERR_NO_COLUMN_FOUND.getMsgForKey(expandOption,
+                                                                                                    entityName));
+      }
+
+      columnDetailList.addAll(expandColumnDetailList);
+
+      freezeSuccessFactorsColumnMetadata(columnDetailList);
+
+      return buildSchema(columnDetailList);
+    } catch (EdmException ee) {
+      throw new SuccessFactorsServiceException(
+        ResourceConstants.ERR_BUILDING_COLUMNS.getMsgForKey(expandOption, entityName), ee);
+    }
+  }
+
+  /**
+   * Finds and builds all the children under the given expanded navigation path.
+   * Example:
+   * Let say the metadata is as follows:
```

Comment:
I would like to understand this logic better, can we go over it in our meeting on Thursday?

Reply:
Yes, we can discuss it in the meeting.